src/internal_rules.jl

```diff
@@ -621,20 +621,20 @@ end
 function EnzymeRules.forward(
         func::Const{typeof(ldiv!)},
         RT::Type,
-        fact::Annotation{C},
-        B,
+        fact::Annotation{<:Cholesky},
+        B::Union{Const{<:AbstractVecOrMat}, DuplicatedNoNeed{<:AbstractVecOrMat}, Duplicated{<:AbstractVecOrMat}, BatchDuplicatedNoNeed{<:AbstractVecOrMat}, BatchDuplicated{<:AbstractVecOrMat}};
```

Comment:
This shouldn't be necessary (and is limiting if different annotations come out). The only uses of B are in a similar ldiv and B inputs are guaranteed valid arguments to ldiv (since that's the only way this could be called).

Reply:
Ah, I was fooled by if-else statements in lines 666-676 - I somehow mixed up `RT` with `B` and assumed that we want `B` to be one of these annotations to ensure that all cases are covered correctly. But
```suggestion
        B::Annotation{<:AbstractVecOrMat};
```
should be fine, shouldn't it? `ldiv!(::Cholesky, x)` is not defined for any other types of `x`:
```julia
julia> methods(ldiv!, Tuple{Cholesky, Any})
# 2 methods for generic function "ldiv!" from LinearAlgebra:
 [1] ldiv!(C::Cholesky{T, <:StridedMatrix{T} where T}, B::StridedVecOrMat{T}) where T<:Union{Float32, Float64, ComplexF64, ComplexF32}
     @ ~/.asdf/installs/julia/1.10.0/share/julia/stdlib/v1.10/LinearAlgebra/src/cholesky.jl:577
 [2] ldiv!(C::Cholesky, B::AbstractVecOrMat)
     @ ~/.asdf/installs/julia/1.10.0/share/julia/stdlib/v1.10/LinearAlgebra/src/cholesky.jl:580
```

Reply:
I'd prefer to leave it unspecified in case Julia APIs or other packages overload the method.

Reply:
```suggestion
        B;
```